src/components/victory-chart.jsx

```diff
@@ -9,27 +9,29 @@ import {VictoryAxis} from "victory-axis";
 class VictoryChart extends React.Component {
   constructor(props) {
     super(props);
-    /*
-       Our use-cases are:
-       1. The user passes in data as an array of {x: 1, y: 2}
-       2. The user provides no x; make it from xMin and xMax
-       3. The user provides x as an array of points; leave it be
-       4. The user provides y as an array of points; leave it be
-       5. The user provides y as a function; use x to generate y
-     */
+
+    // Initialize state
+    this.state = {};
+
+    // Either we get data, or we get X/Y.
+    // `y` will either be none, a function, an array of numbers,
+    // or an array of functions.
     if (this.props.data) {
-      this.state = {
-        data: this.props.data,
-        x: this.props.data.map(row => row.x),
-        y: this.props.data.map(row => row.y)
-      };
+      this.state.data = this.props.data;
     } else {
       this.state = {};
       this.state.x = this.returnOrGenerateX();
       this.state.y = this.returnOrGenerateY();
 
-      const inter = _.zip(this.state.x, this.state.y);
-      const objs = _.map(inter, (obj) => { return {x: obj[0], y: obj[1]}; });
+      const inter = _.map(this.state.y, (y) => _.zip(this.state.x, y));
+
+      const objs = {};
+
+      // Create a map {line-number: dataForLine}
+      _.forEach(inter, (objArray, idx) => {
+        const k = "line-" + idx;
+        objs[k] = (_.map(objArray, (obj) => {return {x: obj[0], y: obj[1]}; }));
+      });
```

Comment:
I think it may look something like this:

```js
const objs = _(inter)
  .map((objArray, idx) => { 
    return [line-" + idx,  _.map(objArray, (obj) => {return {x: obj[0], y: obj[1]}; })];
  })
  .object()
  .value();
```